### ISSUE TITLE
Mattermost panics when enabling a plugin

### ISSUE THREAD

**Issue by SmartHoneybee** [mattermost/mattermost-server#9591]
#### Summary
Mattermost panics if a directory in `./plugins` is (e.g.) not traversable (any error returned by `os.Open` that is not `os.ErrNotExist` should trigger this).

#### Steps to reproduce
* Happens in `v5.3.1`. (Should also happen on `v5.4.0-rc1`.)
* Create a directory called `a_test` in `./plugins`. (Exact name doesn't matter, but has to be lexicographically smaller than other plugins installed.)
* Restrict Mattermost user from traversing it. (E.g. `chown root:root a_test; chmod 776 a_test`.)
* Try to enable or disable a plugin.

#### Expected behavior
Mattermost ignores the directory and its content and changes the plugin's state.

#### Observed behavior (that appears unintentional) 
Activated/deactivated plugins will remain in state: `This plugin is starting.` / `This plugin is stopping.`

Mattermost panics with:
```
goroutine 3604 [running]:
runtime/debug.Stack(0x0, 0x0, 0x107c5a0)
        /usr/local/go/src/runtime/debug/stack.go:24 +0xa7
runtime/debug.PrintStack()
        /usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/mattermost/mattermost-server/vendor/github.com/gorilla/handlers.recoveryHandler.log(0x1441080, 0xc420208e00, 0x1440e00, 0x1c4e3f0, 0x1, 0xc421a532d0, 0x1, 0x1)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/vendor/github.com/gorilla/handlers/recovery.go:89 +0x70
github.com/mattermost/mattermost-server/vendor/github.com/gorilla/handlers.recoveryHandler.ServeHTTP.func1(0x144bda0, 0xc4213ee460, 0x1441080, 0xc420208e00, 0x1440e00, 0x1c4e3f0, 0x1)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/vendor/github.com/gorilla/handlers/recovery.go:74 +0xe5
panic(0x1116600, 0x1bf1fd0)
        /usr/local/go/src/runtime/panic.go:502 +0x229
github.com/mattermost/mattermost-server/app.(*App).EnablePlugin(0xc4200ea580, 0xc421ff6225, 0x4, 0xc421ac0320)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/app/plugin.go:196 +0x1c8
github.com/mattermost/mattermost-server/api4.enablePlugin(0xc42282b500, 0x144bda0, 0xc4213ee460, 0xc42282b400)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/api4/plugin.go:183 +0x175
github.com/mattermost/mattermost-server/web.Handler.ServeHTTP(0xc4200ea580, 0x13350f0, 0x10001, 0x144bda0, 0xc4213ee460, 0xc42282b400)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/web/handlers.go:131 +0x126d
github.com/mattermost/mattermost-server/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420208e00, 0x144bda0, 0xc4213ee460, 0xc42282b400)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/vendor/github.com/gorilla/mux/mux.go:162 +0xed
github.com/mattermost/mattermost-server/vendor/github.com/gorilla/handlers.recoveryHandler.ServeHTTP(0x1441080, 0xc420208e00, 0x1440e00, 0x1c4e3f0, 0x1, 0x144bda0, 0xc4213ee460, 0xc42282b200)
        /home/ubuntu/jenkins/workspace/msr/mattermost-platform-release/src/github.com/mattermost/mattermost-server/vendor/github.com/gorilla/handlers/recovery.go:78 +0xb9
net/http.serverHandler.ServeHTTP(0xc421f03790, 0x144bda0, 0xc4213ee460, 0xc42282b200)
        /usr/local/go/src/net/http/server.go:2694 +0xbc
net/http.(*conn).serve(0xc42297d2c0, 0x144cc60, 0xc4202116c0)
        /usr/local/go/src/net/http/server.go:1830 +0x651
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2795 +0x27b
```

#### Possible fixes

https://github.com/mattermost/mattermost-server/blob/0788cdcadfb5d76b08758f42f01521b45ea76362/model/manifest.go#L254-L257

Set `path = ""`, similar to:

https://github.com/mattermost/mattermost-server/blob/0788cdcadfb5d76b08758f42f01521b45ea76362/model/manifest.go#L279-L281

Additionally, it would be nice, if [`scanSearchPath`](https://github.com/mattermost/mattermost-server/blob/fe9a81208e4d8290df7b8d89bac2d880c045b84b/plugin/environment.go#L72) would return `info.err` in case it is not `nil`.

**Comment by amyblais** [comment]
Thank you for reporting this @SmartHoneybee.

@cpanato Can you help take a look at this one ^

**Comment by cpanato** [comment]
its bug we can fix it.

@SmartHoneybee the `a_test`is a plugin or just an empty folder? do you want to submit the Pr to fix that? otherwise, I can to that

thanks for finding this bug

**Comment by amyblais** [comment]
Opened a Jira ticket here: https://mattermost.atlassian.net/browse/MM-12641.

**Comment by SmartHoneybee** [comment]
Hi @cpanato,

`a_test` is just an empty folder. (But it's content shouldn't matter. More context: https://pre-release.mattermost.com/core/pl/k3fmzaknatyh7xu1wdrye7ofto )

I cannot submit PRs here (CLA).
You submitting a PR would be awesome! Thank you! :slightly_smiling_face: 

Regards,
Bee

**Comment by cpanato** [comment]
@amyblais can you please make a jira ticket? thanks

**Comment by cpanato** [comment]
@SmartHoneybee thanks for this

**Comment by amyblais** [comment]
I close this Issue since the PR for this has been merged. I believe the fix will be included in v5.6 release.
